scipy/signal/_signaltools.py

```diff
@@ -4625,3 +4625,128 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
         sl[axis] = slice(None, None, q)
 
     return y[tuple(sl)]
+
+def envelope(x, N=None, method='analytic'):
+    """
+    Calculate the upper and lower envelopes of a 1-D signal.
+
+    By default, the envelopes of `x` are calculated by the magnitude of its
+    analytical signal.
+
+    Parameters
+    ----------
+    x : array_like
+        The 1-D array holding the signal data for which the envelopes are
+        calculated.
+    N : int
+        If `method` is `'analytic'`, `N` represents the number of Fourier components.
+        If `method` is `'rms'`, `N` is the sliding window length.
+        If `method` is `'peak'`, `N` is the required minimal horizontal distance
+        between neighboring peaks. For `'rms'` and `'peak'` methods, `N` must be
+        specified and must be positive. Default: ``x.shape[0]``
+    method : str {`'analytic'`, `'rms'`, `'peak'`}, optional
+        A string indicating which method to use to calculate the envelopes.
+
+        ``analytic``
+            The envelopes of `x` are calculated by the magnitude of its
+            analytical signal.
+        ``rms``
+            Calculates the upper and lower root-mean-square envelopes of `x`. The
+            mean is determined by a sliding window of `N` samples.
+        ``peak``
+            Calculates the upper and lower peak envelopes of `x`. The envelopes
+            are determined by employing spline interpolation between adjacent
+            local maxima that are at least `N` samples apart.
```

Comment:
I feel like a "Notes" section outlining each of the three approaches would be very useful to guide users in what to pass to `N`. Currently, without looking at the code, I wouldn't be sure how the "number of Fourier components" impacts the result.